### PR TITLE
Add video padding and scaling information to the protocol

### LINKED ISSUE #175: Add video padding and scaling information to the protocol
Allow transferring the information about padding and scaling added to the frame within the protocol.

* source_width, source_height, 
* output_width, output_height, 
* padding_left, padding_top, padding_right, padding_bottom.

Connected to #229

### PR DESCRIPTION
**Pull request by tomskikh**
Connected to https://github.com/insight-platform/Savant/pull/327

### CODE REVIEW

File: savant/gstreamer/metadata.py [hunk 1]
```diff
@@ -48,6 +48,61 @@ def clear(self) -> None:
...
+def transformation_from_dict(d: Dict[str, Any]):
+    if d['type'] == 'initial_size':
+        return InitialSizeTransformation(d['width'], d['height'])
+    elif d['type'] == 'resulting_size':
+        return ResultingSizeTransformation(d['width'], d['height'])
+    elif d['type'] == 'padding':
+        return PaddingTransformation(d['left'], d['top'], d['right'], d['bottom'])
+    elif d['type'] == 'scale':
+        return ScaleTransformation(d['width'], d['height'])
+    else:
```
Comment (abramov-oleg) [inline savant/gstreamer/metadata.py]: Unnecessary `elif` and `else`

File: savant/api/builder.py [hunk 2]
```diff
@@ -172,3 +177,30 @@ def add_tags_to_video_frame(
...
+            transformation['top'],
+            transformation['right'],
+            transformation['bottom'],
+        )
+    elif transformation['type'] == 'scale':
+        return FrameTransformation.scale(
+            transformation['width'],
+            transformation['height'],
+        )
+    else:
```
Comment (abramov-oleg) [inline savant/api/builder.py]: Unnecessary `elif` and `else`

File: savant/gstreamer/metadata.py [hunk 3]
```diff
@@ -48,6 +48,61 @@ def clear(self) -> None:
...
+    def to_dict(self):
+        return {**self._asdict(), 'type': 'padding'}
+
+
+Transformation = Union[
+    InitialSizeTransformation,
+    ResultingSizeTransformation,
+    ScaleTransformation,
+    PaddingTransformation,
+]
```
Comment (abramov-oleg) [inline savant/gstreamer/metadata.py]: Are these types useful?

As I see it, the transformation tuples are built from dicts and are consumed as dicts. Can they just be dicts?

**Review by abramov-oleg — CHANGES_REQUESTED** [review]

File: savant/gstreamer/metadata.py (hunk 1 above)
Comment (tomskikh) [reply savant/gstreamer/metadata.py]: Reworked this PR after merging https://github.com/insight-platform/Savant/pull/367

**Review by abramov-oleg — APPROVED** [review]

**Pull request merged**